src/drivers/net/dwc_gmac/dwc_gmac.c

```diff
@@ -620,32 +627,53 @@ static int dwc_init(void) {
 	rx_len = RX_DESC_QUANTITY * sizeof(struct dma_extended_desc);
 	tx_len = TX_DESC_QUANTITY * sizeof(struct dma_extended_desc);
 
-	dwc_priv.rxdesc_ring_paddr = periph_memory_alloc(rx_len);
-	memset(dwc_priv.rxdesc_ring_paddr, 0, rx_len);
+	dwc_priv.rxdesc_ring = dma_alloc_coherent(NULL, rx_len,  &dwc_priv.rxdesc_ring_dma, 0);
+	if (NULL == dwc_priv.rxdesc_ring) {
+		log_error("Couldnt alloc periph mem for rxdesc ring");
+		res = -ENOMEM;
+		goto err_out_etherdev_free;
+	}
+
+	dwc_priv.txdesc_ring = dma_alloc_coherent(NULL, tx_len, &dwc_priv.txdesc_ring_dma, 0);
+	if (NULL == dwc_priv.txdesc_ring) {
+		log_error("Couldnt alloc periph mem for txdesc ring");
+		res = -ENOMEM;
+		goto err_out_rxdesc_free;
+	}
+
+	memset(dwc_priv.rxdesc_ring, 0, rx_len);
 	/* TODO: this flush is neccessary even after we write to memory
 	 * which was allocated as cached */
-	dcache_flush(dwc_priv.rxdesc_ring_paddr, rx_len);
+	dcache_flush((void *)dwc_priv.rxdesc_ring_dma, rx_len);
```

Comment:
Do we assume that value passed to `dcache_flush()`/`dcache_inval()` is physical or virtual address?
Looks like it's incorrect to pass dma address either way.

Do we really need to flush anything if memory is coherent?